### PR TITLE
Add a little shell example.

### PR DESCRIPTION
**Pull request by thejpster**
You can walk around a disk image and do directory listings. I had to make some more things 'Debug' so I could print the filesystem state. Also our handling of ".." was not right so I fixed that.

### CODE REVIEW

**Review by eldruin — COMMENTED** [review]
Thanks!
Could you add an entry to the changelog and have a look at the CI?

**Comment by thejpster** [comment]
@eldruin  added as requested. Should be ready now.

File: examples/shell.rs [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+impl Context {
+    fn current_path(&self) -> Vec<String> {
+        let Some(s) = &self.volumes[self.current_volume] else {
+            return vec![];
+        };
+        s.path.clone()
+    }
+
+    fn process_line(&mut self, line: &str) -> Result<(), Error<std::io::Error>> {
+        if line == "help" {
```
Comment (ryan-summers) [inline examples/shell.rs:34]: FYI, https://docs.rs/logos/latest/logos/ is really nice for things like this :)

**Review by ryan-summers — APPROVED** [review]

File: examples/shell.rs (hunk 1 above)
Comment (thejpster) [reply examples/shell.rs:34]: Interesting! I'll check it out.

**Pull request merged**